grazer/src/main/java/edu/asu/diging/grazer/core/fileupload/service/impl/FileUploadServiceImpl.java

```
@@ -0,0 +1,96 @@
+import org.springframework.web.multipart.commons.CommonsMultipartFile;
+
+import edu.asu.diging.grazer.core.domain.impl.FileImpl;
+import edu.asu.diging.grazer.core.domain.impl.FileTransformationImpl;
+import edu.asu.diging.grazer.core.fileupload.db.impl.FileUploadDatabaseConnection;
+import edu.asu.diging.grazer.core.fileupload.service.IFileUploadService;
+
+@Service
+@PropertySource(value = "classpath:config.properties")
+public class FileUploadServiceImpl implements IFileUploadService {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    
+    @Autowired
+    private FileUploadDatabaseConnection connection;
+    
+    @Autowired
+    private FileImpl file;
+    
+    @Autowired
+    private Environment env;
+    
+    public void uploadFiles(HashMap<String, byte[]> filesMap) {
+        
+        for(String key: filesMap.keySet()) {
+            try {      
+                int num = 1;
+                
+                String dir = env.getProperty("transformation.file.dir");
+                String serverFileName = dir + File.separator + key;
+                String extension = FilenameUtils.getExtension(serverFileName);
+                String serverFileNameWithoutExt = FilenameUtils.removeExtension(serverFileName);
+                
+                File serverFile = new File(serverFileName);
+                while(serverFile.exists()) {
+                    serverFileName = serverFileNameWithoutExt + (num++) + "." + extension;
+                    serverFile = new File(serverFileName);
+                }
+
+                BufferedOutputStream stream = new BufferedOutputStream(new FileOutputStream(serverFile));
+                stream.write(filesMap.get(key));
+                stream.close();
+                logger.info("Server File Location = " + serverFile.getAbsolutePath());
+            } catch(IOException e) {
+                logger.error("You failed to upload " + key + e.getMessage());
```

Don't swallow the exception.